src/formatDate.js

```diff
@@ -50,7 +50,39 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  let newDate = '';
+  const newArray = [];
+
+  const oldDate = date.split(fromFormat[fromFormat.length - 1]);
+
+  for (let i = 0; i < toFormat.length - 1; i++) {
+    if (toFormat[i] === 'DD' || toFormat[i] === 'MM') {
+      for (let j = 0; j < fromFormat.length - 1; j++) {
+        if (fromFormat[j] === toFormat[i]) {
+          newArray[i] = oldDate[j];
+        }
+      }
+    }
+
+    if (toFormat[i] === 'YY' || toFormat[i] === 'YYYY') {
+      for (let j = 0; j < fromFormat.length - 1; j++) {
+        if (fromFormat[j] === 'YY' && toFormat[i] === 'YYYY') {
+          const year = parseInt(oldDate[j], 10);
+
+          newArray[i] = (year < 30 ? '20' : '19')
+          + (year < 10 ? '0' : '') + year;
+        } else if (fromFormat[j] === 'YYYY' && toFormat[i] === 'YY') {
+          newArray[i] = oldDate[j].substring(2);
+        } else if (fromFormat[j] === toFormat[i]) {
+          newArray[i] = oldDate[j];
```

Comment:
cause you have the same check here
<img width="428" alt="image" src="https://github.com/mate-academy/js_format_date_advanced/assets/70220667/c5ab02e3-2ab2-44a5-a813-a4a95b54babe">